### PR TITLE
Superclass

### PR DESCRIPTION
**Pull request by lumi-a**
*(The sound of several plastic bottles falling over can be heard in the background)*

I am *so* tired and should finish up for today, but still wanted to post this PR before going to sleep. In the morning, I'll add a description as to what the heck it does, but it can already be merged, even though it doesn't do everything I want it to yet. It'll probably introduce many new bugs, but they will be easier to fix moving forward.

In short: This introduces a hierarchy between different classes, significantly cutting down on code-duplication, making future code easier to write, and making the main-loop so, *so* much easier to read.
> *"Once we are done with this, it will be actually a pleasure to add stuff to this code."*

### CODE REVIEW

**Comment by lumi-a** [comment]
- The camera now zooms out further when the ship has high speed. Try bumping into a planet, it even zooms smoothly despite the rapid change of the ship's velocity!
- The camera no longer stops at the universe's edge, but that can be reimplemented in the future.

**Comment by lumi-a** [comment]
- Everything is now drawn using anti-aliasing, meaning edges are finally smooth. Compare:

    image

    There is still some coordinate-overflow bug which causes a crash a few seconds into the game, but I'll fix that soon by only drawing shapes that actually are on-screen.

**Comment by lumi-a** [comment]
- Alright, overflow-bug is fixed 💅

**Comment by lumi-a** [comment]
- `Square`s have been renamed to `Area`s. The game-destination Area has been removed, as I don't yet know how to handle that using the new class.
- Text-rendering is now done in `Universe`
- As a result, the main-loop is now super-concise, fitting on a single screen.
